include/toolchain/xcc.h

```diff
@@ -52,4 +52,10 @@
 #define __builtin_umulll_overflow(a, b, output) \
 	({ *output = (a) * (b); false; })
 
+#ifdef ALWAYS_INLINE
+/* XCC said it needs "gnu_inline" */
+#undef ALWAYS_INLINE
+  #define ALWAYS_INLINE inline __attribute__((always_inline)) __attribute__((gnu_inline))
+#endif
```

Comment:
This bit needs to go in include/toolchain somewhere.  The idea is that the compiler integration there defines ALWAYS_INLINE in a way that makes sense for it.

Reply:
it is in include/toolchain already.

Reply:
Uh... so it is.  Never mind then :)